openedx/core/djangoapps/course_groups/partition_generator.py

```diff
@@ -0,0 +1,72 @@
+"""
+The team dynamic partition generation to be part of the
+openedx.dynamic_partition plugin.
+"""
+import logging
+
+from django.utils.translation import gettext_lazy as _
+from openedx.core.djangoapps.course_groups.flags import CONTENT_GROUPS_FOR_TEAMS
+
+from xmodule.partitions.partitions import UserPartition, UserPartitionError
+from xmodule.services import TeamsConfigurationService
+
+log = logging.getLogger(__name__)
+
+MINIMUM_DYNAMIC_TEAM_PARTITION_ID = 51
```

Comment:
I understand that every partition needs a numeric id, and that we must assign them carefully as to not conflict with one another.

Can you help me understand how 51 was chosen as the minimum here, and how you are sure that this will not collide with other partition IDs?

Reply:
@mariajgrimaldi Thanks for that context. I dove a bit deeper and wrote up the list of a partition IDs that are in use right now: https://openedx.atlassian.net/wiki/spaces/~6193e357d5986c006a6713bb/pages/4146823192/Notes+on+Content+Groups+User+Partitions+and+Teams#Known-User-Partitions. It looks like 51 is taken by the ContentTypeGating partition, so the range we'd need to use here is 52-99.

Even so, that creates a somewhat arbitrary limit of 47 team-set-partitions per course. Is that limitation OK? You would need to add some checks to ensure that team-set partitions don't rise above id 99, and have some fallback behavior for team-sets above the limit.

Alternatively, it would take some refactoring, but I think in the end it would be simpler to use the 100->MAX_INT range. Both the Content Group partition and the Experiment Group Partitions (aka "random" partitions) share that range, and they avoid conflicts by calling:
```python
            generate_int_id(
                MINIMUM_STATIC_PARTITION_ID,
                MYSQL_MAX_INT,
                GroupConfiguration.get_used_ids(self.course),  # avoid existing Content & Experiment partitions
            )
```
Of course, you would need to extract `GroupConfiguration.get_used_ids(...)` out into a new utility function that returned *all* the dynamic partitions: both those from GroupConfiguration and those from the team-sets.

Reply:
We were aware of the limitation of the number of team sets, and because of the use case of in-campus learning it wasn't a blocker for us. I would prefer to try solving this limitation instead of having a fallback behavior, so I tested your suggestion, and it worked great!